### ISSUE TITLE
Problem: Jobs initiated twice

### ISSUE THREAD

**Issue by overture8** [c-rack/quantum-elixir#293]
I have a cluster of 3 nodes and I'm using a `Quantum.RunStrategy.Random` strategy. This didn't happen for me before upgrading to 2.2.1.

Here is my config:

```elixir
config :my_app, MyApp.Scheduler,
  global: true,
  jobs: [
    [schedule: "0 7 * * 1",
     timezone: :utc,
     run_strategy: {Quantum.RunStrategy.Random, :cluster},
     task: {Core, :job1, []}],

    [schedule: "*/30 * * * *",
     timezone: :utc,
     run_strategy: {Quantum.RunStrategy.Random, :cluster},
     task: {Core, :job2, []}],

    [schedule: "*/35 * * * *",
     timezone: :utc,
     run_strategy: {Quantum.RunStrategy.Random, :cluster},
     task: {Core, :job3, []}]
  ]
```

The job that this seemed to happen on was "job1". It was due to process at 7am on Monday. However, it seemed to process all jobs twice.

From looking at my logs, here are the times that the nodes were processing jobs:

* **Node1**: 7:00am to 7:02am
* **Node2**: 7:00am to 7:02am
* **Node3**: 6:49am to 6:51am

The Node3 times really confused me. Have you any suggestion on how I could investigate this more so I can get to the bottom of this?

**Comment by maennchen** [comment]
@overture8 Are you sure that all of the nodes are clustered correctly? Are the times in sync between the nodes?

**Comment by overture8** [comment]
@maennchen hmm, actually - it does seem that the time on Node3 is ten minutes behind the other two nodes (sorry I should have checked that first).

Regarding whether the nodes are clustered correctly - they all seem to be aware of each other if I do `Node.list` at the console. However, I've done another deployment since the issue happened - so maybe it was just an issue with the cluster on that deployment.

So, if Node1 and Node2 clustered correctly and Node3 didn't join for some reason - would this result in the job being ran twice?

**Comment by maennchen** [comment]
@overture8 

## TL;DR
All nodes start the quantum scheduler on startup. If the cluster already contains a scheduler, this one will be linked instead of a new scheduler. If two schedulers are running this will also mean that the jobs are executed twice.

## Details

Depending on the `global` config flag, the name for the task stages supervisor is chosen. ([See here](https://github.com/c-rack/quantum-elixir/blob/master/lib/quantum.ex#L33))

The name is then picked up by the task stages supervisor. ([See here](https://github.com/c-rack/quantum-elixir/blob/master/lib/quantum/task_stages_supervisor.ex#L14))

How the name works is defined by the [Name Registration](https://hexdocs.pm/elixir/GenServer.html#module-name-registration) of GenServers.

If the name is `{:global, :some_name}`, the `start_link` will either return:

* `{:ok, pid}` => The server was started. (No other server with the same name exists in the cluster.)
* `{:error, {:already_started, pid}}` => The server could not be started because it already exists in the cluster. (In this case we link the `pid`.)
* `{:error, _}` => Something else failed.

## Possible Explanation

You started two independent clusters. (somehow the cluster linking did not work as expected) Therefore now Node 1 & 2 have one scheduler and Node 3 has its own scheduler. They both ran the tasks. Since the clock on Node 3 was not correct, it also didn't run the task at the appropriate time.

**Comment by maennchen** [comment]
@overture8 This should explain the issue. If you however still experience the issue after setting the right times and clustering them correctly, please reopen the issue.

**Comment by overture8** [comment]
Many thanks for your help - much appreciated!
